packages/commerce-sdk-react/src/auth/index.test.ts

```diff
@@ -246,4 +223,31 @@ describe('Auth', () => {
             expect(helpers.loginGuestUser).toBeCalled()
         })
     })
+    test('running on the server uses a shared context memory store', async () => {
+        const refreshTokenGuest = 'guest'
+
+        // Mock running on the server so shared context storage is used.
+        // @ts-expect-error read-only property
+        utils.onClient = () => false
+
+        // Create a new auth instance and set its guest token.
+        const authA = new Auth({...config, siteId: 'siteA'})
+        // @ts-expect-error private method
+        authA.set('refresh_token_guest', refreshTokenGuest)
+        // @ts-expect-error private property
+        expect([...authA.stores['memory'].map.keys()]).toEqual([`siteA_cc-nx-g`])
+
+        // Create a second auth instance and ensure that its memory store has previous
+        // guest tokens set from the first store (this emulates a second lambda request.)
+        const authB = new Auth({...config, siteId: 'siteB'})
+        // @ts-expect-error private method
+        authB.set('refresh_token_guest', refreshTokenGuest)
+
+        // @ts-expect-error private property
+        expect([...authB.stores['memory'].map.keys()]).toEqual([`siteA_cc-nx-g`, `siteB_cc-nx-g`])
+
+        // Set mock value back to expected.
+        // @ts-expect-error read-only property
+        utils.onClient = () => true
```

Comment:
I believe jest has util functions to restore mocks, I think it's `jest.restoreAllMocks()` (or maybe `jest.resetAllMocks()`).

We could do that in afterEach to DRY it.
```
afterEach(() => {
  jest.restoreAllMocks()
})
```